src/Dependencies/PooledObjects/ArrayBuilder.cs

```diff
@@ -569,6 +569,14 @@ public void AddRange<U>(ArrayBuilder<U> items, Func<U, T> selector)
             }
         }
 
+        public void AddRange<U>(ImmutableArray<U> items, Func<U, T> selector)
+        {
```

Comment:
Should we add something like the following?

```cs
EnsureCapacity(Count + items.Length);
```

Reply:
Yup, that seems good. I'll leave the other overloads as-is though.